Monika After Story/game/script-python.rpy

```diff
@@ -0,0 +1,1205 @@
+    m 3eua "These versions are {b}incompatible{/b} with each other because the changes added in Python3 fixed many fundamental design flaws in Python2."
+    m "Even though this introduced a split in the Python community,{w} it's generally agreed that both versions of the language have their own strengths and weaknesses."
+    m 3eub "I'll tell you about those differences in another lesson."
+
+    m 1eua "This mod runs on a RenPy version that uses Python2, so I won't be talking about Python3 too often."
+    m 1hua "But I'll mention it when it's appropriate."
+
+    m 3eua "That's my lesson for today."
+    m 1hua "Thanks for listening!"
+    return
+
+###############################################################################
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_ptod_tip002",
+            category=["python tips"],
+            prompt="Types",
+            conditional="store.mas_ptod.has_day_past_tip(3)",
+            action=EV_ACT_POOL
+        )
+    )
+
+# PREREQS:
+#   interpreted language (tip 3)
+label monika_ptod_tip002:
+    $ tip_ev = mas_getEV("monika_ptod_tip002")
+    if tip_ev.last_seen is None:
+        m 1eua "In most programming languages, data that can be changed or modified by the program has a {i}type{/i} associated with it."
+        m 3eua "For example, if some data should be treated as a number, then it will have a numeric type. If some data should be treated as text, then it will have a string type."
+        m "There are many types in Python, but today we'll talk about the more basic, or primitive ones."
+
+    $ store.mas_ptod.rst_cn()
+    $ local_ctx = dict()
+    show monika at t22
+    show screen mas_py_console_teaching
+
+    ### numbers
+    m 1eua "Python has two types to represent numbers:{w} {i}integers{/i}, or {b}ints{/b},{w} and {i}floats{/i}."
+
+    ## integers
+    m 1eua "Integers are used to represent whole numbers; basically anything that isn't a decimal."
+
+    call mas_wx_cmd("type(-22)", local_ctx)
+    call mas_wx_cmd("type(0)", local_ctx)
+    call mas_wx_cmd("type(-1234)", local_ctx)
+    call mas_wx_cmd("type(42)", local_ctx)
+
+    ## floats
+    m 1eub "Floats are used to represent decimals."
+    show monika 1eua
+
+    call mas_wx_cmd("type(0.14)", local_ctx)
+    call mas_wx_cmd("type(9.3)", local_ctx)
+    call mas_wx_cmd("type(-10.2)", local_ctx)
+
+    ### strings
+    m 1eua "Text is represented with {i}string{/i} types."
+    m "Anything surrounded in single quotes (') or double quotes (\") are strings."
+    m 3eub "For example:"
+    show monika 3eua
+    
+    call mas_wx_cmd("type('This is a string in single quotes')", local_ctx)
+    call mas_wx_cmd('type("And this is a string in double quotes")', local_ctx)
+
+    m 1eksdlb "Even though these say {i}unicode{/i}, that is basically the same as a string and can be treated that way for our purposes."
```

Comment:
"I know the terminal says {i}unicode{/i}, but for what we're doing right now, it basically is the same thing."

Reply:
gonna put `interpreter` since its not really a terminal.
Also gonna drop the right now since strings are all going to be unicode in future lessons, but we're going to treat them as strings